agent/src/agent/pipeline/pipeline.py

```diff
@@ -101,8 +101,8 @@ def config_changed(self) -> bool:
     def set_config(self, config: dict):
         self._previous_config = deepcopy(self.config)
         self._previous_override_source = deepcopy(self.override_source)
-        self.override_source = config.pop(self.OVERRIDE_SOURCE, {})
-        self.config = config
+        self.config = deepcopy(config)
+        self.override_source = self.config.pop(self.OVERRIDE_SOURCE, {})
```

Comment:
why do you make a deep copy here and why don't you pop override source from it first?

Reply:
Because `config.pop(self.OVERRIDE_SOURCE, {})` changes the `config` object and we need to create a pipeline with this object two times - for preview and later for actual pipeline creation. So after we create a test pipeline `config` object doesn't have `self.OVERRIDE_SOURCE` property anymore and we need it to create the pipeline and save it to db

Reply:
got it, now I see you pop from self.config so that's perfectly fine :)